### PR TITLE
Have `extract_local_sensitivities` return a 2D subarray

### PR DESCRIPTION
**Pull request by jlumpe**
The current return type for the sensitivity matrix (Array of Arrays) isn't too easy to work with and also copies/allocates memory unnecessarily. This isn't backwards compatible, so I added an additional set of methods taking an `ismatrix::Bool` argument to opt in to the behavior.

### CODE REVIEW

File: src/local_sensitivity.jl [hunk 1]
```diff
@@ -125,24 +125,40 @@ function ODELocalSensitivityProblem(f,args...;kwargs...)
...
+extract_local_sensitivities(sol, i::Integer) = _extract(sol, sol[i], Val{false}())
+extract_local_sensitivities(sol, i::Integer, asmatrix::Bool) = _extract(sol, sol[i], Val{asmatrix}())
+extract_local_sensitivities(sol, t) = _extract(sol, sol(t), Val{false}())
+extract_local_sensitivities(sol, t, asmatrix::Bool) = _extract(sol, sol(t), Val{asmatrix}())
+extract_local_sensitivities(tmp, sol, t) = _extract(sol, sol(tmp, t), Val{false}())
+extract_local_sensitivities(tmp, sol, t, asmatrix::Bool) = _extract(sol, sol(tmp, t), Val{asmatrix}())
+
+
+# Get ODE u vector and sensitivity values from sensitivity problem u vector
+function _extract(sol, su::Vector, asmatrix::Val)
```
Comment (ChrisRackauckas) [inline src/local_sensitivity.jl]:
```suggestion
function _extract(sol, su::Vector, asmatrix::Val = Val(false))
```

**Comment by jlumpe** [comment]
I could use some help resolving some type inference issues here. It seems to me that if I use `asmatrix=true` within a wrapper function that constant propagation should ensure the correct method is called, but it doesn't seem to be working.

**Comment by ChrisRackauckas** [comment]
Yeah, I'm surprised constant prop doesn't kick in here, so I'd just take a Val

**Comment by jlumpe** [comment]
Done, added methods for both.

**Comment by jlumpe** [comment]
Should this be ready to merge?

**Comment by ChrisRackauckas** [comment]
Yes indeed. Thanks!

**Pull request merged**